tests/common.go

```diff
@@ -291,7 +413,163 @@ func ValidateVolumes(ctx *scheduler.Context, errChan ...*chan error) {
+			err = Inst().V.ValidateCreateGroupSnapshotUsingPxctl()
+			if err != nil {
+				expect(err.Error()).To(contain(errPureFileSnapshotNotSupported.Error()))
+			}
+		})
+	})
+}
+
+// ValidateResizeFBPVC is the ginkgo spec for validating resize of volumes
+func ValidateResizeFBPVC(ctx *scheduler.Context, errChan ...*chan error) {
+	context("For validation of an resizing pvc", func() {
+		var err error
+		Step(fmt.Sprintf("inspect %s app's volumes", ctx.App.Key), func() {
+			appScaleFactor := time.Duration(Inst().GlobalScaleFactor)
+			err = Inst().S.ValidateVolumes(ctx, appScaleFactor*defaultTimeout, defaultRetryInterval, nil)
+			processError(err, errChan...)
+		})
+
+		Step(fmt.Sprintf("validating resizing pvcs"), func() {
+			err = Inst().S.ResizePureVolumes(ctx)
+			if err != nil {
+				expect(err).ToNot(haveOccurred())
+			}
+		})
+	})
+}
+
+// ValidateVolumeStatsticsDynamicUpdate is the ginkgo spec for validating dynamic update of byteUsed statstic for pure volumes
+func ValidateVolumeStatsticsDynamicUpdate(ctx *scheduler.Context, errChan ...*chan error) {
+	context("For validation of an resizing pvc", func() {
+		var err error
+		Step(fmt.Sprintf("inspect %s app's volumes", ctx.App.Key), func() {
+			appScaleFactor := time.Duration(Inst().GlobalScaleFactor)
+			err = Inst().S.ValidateVolumes(ctx, appScaleFactor*defaultTimeout, defaultRetryInterval, nil)
+			processError(err, errChan...)
+		})
+		var vols []*volume.Volume
+		Step(fmt.Sprintf("get %s app's pure volumes", ctx.App.Key), func() {
+			vols, err = Inst().S.GetPureVolumes(ctx)
+			processError(err, errChan...)
+		})
+		byteUsedInitial, err := Inst().V.ValidateGetByteUsedForVolume(vols[0].ID, make(map[string]string))
+		fmt.Println(fmt.Sprintf("initially the byteUsed is %v", byteUsedInitial))
+		// get the pod for this pvc
+		pods, err := Inst().S.GetPodsForPVC(vols[0].Name, vols[0].Namespace)
+
+		// write to the FB volume
+		cmdArgs := []string{"exec", "-it", pods[0].Name, "-n", pods[0].Namespace, "--", "bash","-c", "dd bs=521 count=1048576 </dev/urandom > "+ getMountPath(pods[0].Namespace)+ "/myfile"}
+		err = osutils.Kubectl(cmdArgs)
+		processError(err, errChan...)
+		// wait until the backends size is reflected before making the REST call
+		time.Sleep(20 * time.Second)
+
+		byteUsedafter, err := Inst().V.ValidateGetByteUsedForVolume(vols[0].ID, make(map[string]string))
+		fmt.Println(fmt.Sprintf("after writing random bytes to the file the byteUsed is %v", byteUsedafter))
+		err = expectedSizechange(byteUsedafter - byteUsedInitial)
+		expect(err).NotTo(haveOccurred())
+
+	})
+}
+
+func expectedSizechange (sizeChangeInBytes uint64) error {
+	if sizeChangeInBytes > 531 * 1048576 && sizeChangeInBytes < 511 * 1048576 {
+		return errUnexpectedSizeChangeAfterFBIO
+	}
+	return nil
+}
+// getMountPath checkts the podname prefix, and finds the mountpath.
+//	unfortunately, the mountpath for PVCs all vary among applications, so Im getting the hardcoded value here
```

Comment:
Is there a way we could find it by running the `mount` command and looking for NFS mounts? Or is that too slow/this doesn't have access to running commands?